src/libxtp/threecenter_gwbse.cc

```diff
@@ -40,17 +43,33 @@ void TCMatrix_gwbse::Initialize(int basissize, int mmin, int mmax, int nmin,
       _mtotal, Eigen::MatrixXd::Zero(_ntotal, _basissize));
 }
 
+// Maximum number of simultaneous streams running in the GPU
+void TCMatrix_gwbse::SetGPUStreams(int streams) { _max_gpu_streams = streams; }
+
 /*
  * Modify 3-center matrix elements consistent with use of symmetrized
- * Coulomb interaction.
+ * Coulomb interaction using either CUDA or Openmp
  */
 void TCMatrix_gwbse::MultiplyRightWithAuxMatrix(const Eigen::MatrixXd& matrix) {
 
-#pragma omp parallel for
-  for (int i_occ = 0; i_occ < _mtotal; i_occ++) {
-    Eigen::MatrixXd temp = _matrix[i_occ] * matrix;
-    _matrix[i_occ] = temp;
+  // Try to run the operation in a Nvidia GPU, otherwise is the default Openmp
```

Comment:
I am not sure, we can leave this for the moment, but I really from a performance standpoint do not like that when CUDA runs openmp does nothing

Reply:
solved